src/lib/client.ts

```diff
@@ -1,30 +1,48 @@
 import { ApiService } from './api/api.service';
+import { PermSdkError } from './error';
 import { AuthResource } from './resources/auth.resource';
 
 export interface PermanentConstructorConfigI {
   sessionToken: string;
   mfaToken: string;
-  archiveId: number;
+  archiveNbr: string;
   apiKey: string;
+  baseUrl?: string;
 }
 
 export class Permanent {
   private apiKey: string;
   private sessionToken: string;
   private mfaToken: string;
-  private archiveId: number;
+  private archiveNbr: string;
 
-  private api: ApiService;
+  public api: ApiService;
 
   public auth: AuthResource;
   constructor(config: PermanentConstructorConfigI) {
-    const { sessionToken, mfaToken, archiveId, apiKey } = config;
+    const { sessionToken, mfaToken, archiveNbr, apiKey, baseUrl } = config;
+
+    if (!sessionToken) {
+      throw new PermSdkError('Missing sessionToken in config');
+    }
+
+    if (!mfaToken) {
+      throw new PermSdkError('Missing mfaToken in config');
+    }
+
+    if (!archiveNbr) {
+      throw new PermSdkError('Missing archiveNbr in config');
+    }
```

Comment:
These are a good first step towards input validation!

I'll note that we're still not validating the type of the input, only its presence - we could call this from JavaScript with `new Permanent({apiKey: 1, ...etc})` and these checks would pass. But still, this is good progress towards #7 - thanks!

Also, specifically for this parameter, is it safe to say that `0` is never a valid `archiveNbr`?

Reply:
Safe to say, yes! Valid `archiveNbr` will always be a string.

#7 will be great, coming soon!